api/features/workflows/core/models.py

```diff
@@ -181,15 +183,25 @@ def email_subject(self):
 
     @hook(AFTER_CREATE, when="committed_at", is_not=None)
     @hook(AFTER_SAVE, when="committed_at", was=None, is_not=None)
-    def schedule_audit_log_creation_task_for_feature_state_going_live(self):
-        now = timezone.now()
-        feature_states = list(
-            self.feature_states.filter(live_from__gt=now).order_by("live_from")
-        )
-        for feature_state in feature_states:
-            create_feature_state_went_live_audit_log.delay(
-                delay_until=feature_state.live_from, args=(feature_state.id,)
-            )
+    def create_audit_log_for_related_feature_state(self):
+        for feature_state in self.feature_states.all():
+            if self.committed_at < feature_state.live_from:
+                create_feature_state_went_live_audit_log.delay(
+                    delay_until=feature_state.live_from, args=(feature_state.id,)
+                )
+            else:
+                message = FEATURE_STATE_UPDATED_BY_CHANGE_REQUEST_MESSAGE % (
+                    feature_state.feature.name,
+                    feature_state.change_request.title,
+                )
+                AuditLog.objects.create(
+                    related_object_id=feature_state.id,
+                    related_object_type=RelatedObjectType.FEATURE_STATE.name,
+                    environment=feature_state.environment,
+                    project=feature_state.environment.project,
+                    log=message,
+                    is_system_event=True,
```

Comment:
Hmm... I guess this is an interesting question. Is it a system event or an action performed by the change request creator or the CR committer?

Reply:
Yeah, good question, I am leaning towards the request creator? 
Note: we also need to change this for feature state going live (it uses system event)